### PR TITLE
Random UI Improvements

### PR DESCRIPTION
**Pull request by acikek**
- Improved UI debug mode by appending `UIElementStyle` information and adding a yellow outline for hovered elements
- Renamed `TextAlignment` to `UIAlignment` and added fields to `UIAnchors` - now they should *really* never be constructed manually, possibly due for a future cleanup
- Removed text alignments from `UIElementText` (made no sense anyways)
- Moved `Selected` from `UIInputLabel` onto `UIElement` natively, with on select/deselect callbacks
- Added directional navigation callbacks to `UIElement` with the vision of unifying keyboard/gamepad inputs in the future
- `UIListGroup` now supports all expansion directions and is dpeendent on the supplied position anchor

### CODE REVIEW

File: FGEGraphics/UISystem/UIPositionHelper.cs [hunk 1]
```diff
@@ -190,10 +190,6 @@ public UIPositionHelper GetterXY(Func<int> x, Func<int> y)
         return this;
     }
 
-    /// <summary>Sets getter X and Y values using a <see cref="Location"/>.</summary>
-    /// <returns>This object.</returns>
-    public UIPositionHelper GetterXY(Func<Location> location) => GetterXY(() => (int) location().X, () => (int) location().Y);
```
Comment (mcmonkey4eva) [inline FGEGraphics/UISystem/UIPositionHelper.cs:195]: why is this removed?

File: FGEGraphics/UISystem/UIElement.cs [hunk 2]
```diff
@@ -617,9 +668,15 @@ public virtual void Destroy()
...
-        string pos = $"^r^t^0^h^o^e^7Position: ^3({X}, {Y}) ^&| ^7Dimensions: ^3({Width}w, {Height}h) ^&| ^7Rotation: ^3{LastAbsoluteRotation}";
-        string state = $"^7Enabled: ^{(Enabled ? "2" : "1")}{Enabled} ^&| ^7Hovered: ^{(Hovered ? "2" : "1")}{Hovered} ^&| ^7Pressed: ^{(Pressed ? "2" : "1")}{Pressed}";
-        return [type, pos, state];
+        List<string> info = [];
+        info.Add($"^t^0^h^5^u{GetType()}");
+        info.Add($"^r^t^0^h^o^e^7Position: ^3({X}, {Y}) ^&| ^7Dimensions: ^3({Width}w, {Height}h) ^&| ^7Rotation: ^3{LastAbsoluteRotation}");
+        info.Add($"^7Enabled: ^{(Enabled ? "2" : "1")}{Enabled} ^&| ^7Hovered: ^{(Hovered ? "2" : "1")}{Hovered} ^&| ^7Pressed: ^{(Pressed ? "2" : "1")}{Pressed}");
+        if (ElementInternal.Styles.Count > 0)
+        {
+            List<string> styleNames = ElementInternal.Styles.Select(style => style.Name is string name ? $"^{(style == ElementInternal.CurrentStyle ? "3" : "7")}{name}" : "^1unnamed").ToList();
```
Comment (mcmonkey4eva) [inline FGEGraphics/UISystem/UIElement.cs]: `style.Name is string name` it sure is a string yes. is this meant to be a null check?

Comment (mcmonkey4eva) [reply FGEGraphics/UISystem/UIElement.cs]: also `ToList` -- is VS not highlighting that to autoreplace with `[.. x]` syntax?

File: FGEGraphics/UISystem/UIElement.cs [hunk 3]
```diff
@@ -617,9 +668,15 @@ public virtual void Destroy()
     /// <summary>Returns debug text to add to <see cref="ViewUI2D.InternalData.DebugInfo"/>.</summary>
     public virtual List<string> GetDebugInfo()
     {
-        string type = $"^t^0^h^5^u{GetType()}";
-        string pos = $"^r^t^0^h^o^e^7Position: ^3({X}, {Y}) ^&| ^7Dimensions: ^3({Width}w, {Height}h) ^&| ^7Rotation: ^3{LastAbsoluteRotation}";
-        string state = $"^7Enabled: ^{(Enabled ? "2" : "1")}{Enabled} ^&| ^7Hovered: ^{(Hovered ? "2" : "1")}{Hovered} ^&| ^7Pressed: ^{(Pressed ? "2" : "1")}{Pressed}";
-        return [type, pos, state];
+        List<string> info = [];
```
Comment (mcmonkey4eva) [inline FGEGraphics/UISystem/UIElement.cs]: (minor) `new(4)` can save some reallocations here

File: FGEGraphics/UISystem/UIElement.cs [hunk 4]
```diff
@@ -574,6 +588,43 @@ public virtual void MouseLeftUpOutside()
     {
     }
 
+    // TODO: Gamepad input
```
Comment (mcmonkey4eva) [inline FGEGraphics/UISystem/UIElement.cs]: this `// TODO` should be inside `TickInput` I think

File: FGEGraphics/UISystem/UIPositionHelper.cs (hunk 1 above)
Comment (acikek) [reply FGEGraphics/UISystem/UIPositionHelper.cs:195]: I don't see a reason for it to exist, was only used in `UIElementText` w.r.t. text alignment implementation. Forgot to remove the constant version

File: FGEGraphics/UISystem/UIElement.cs (hunk 2 above)
Comment (acikek) [reply FGEGraphics/UISystem/UIElement.cs]: > `style.Name is string name` it sure is a string yes. is this meant to be a null check?

I'm used to nullables being enabled wherein this is how you use nullable objects that you null check for. Can just switch it out for a `is not null` if that's cleaner.

Comment (mcmonkey4eva) [reply FGEGraphics/UISystem/UIElement.cs]: Yes use `is not null`

Even when `is string` _works_ as a null check, it is not _clear_ as a null check - give two relatively equal choices, the one that's clearer about intent is definitely better

File: FGEGraphics/UISystem/UIPositionHelper.cs (hunk 1 above)
Comment (mcmonkey4eva) [reply FGEGraphics/UISystem/UIPositionHelper.cs:195]: It's a nice convenience/utility - remember that FGE is a library, not an exclusive user. ie it very intentionally contains many unused public methods. Library methods should only be removed if there's an active reason to remove them (eg no longer fits design, or encourages bad patterns, or etc).
